lessons/07-mcp/README.md

```diff
@@ -0,0 +1,482 @@
+
+![Run the tool](https://softchris.github.io/mcp-workshop/assets/images/ran-tool-271ecd5f84a457462e59789b579ee9d4.png)
+
+**Scipio**: "Interesting, I wish I could test my strategies like this. Seems very useful."
+
+**You**: "Right, it really does**! I can see how this would be useful for testing and debugging."
+
+## Creating a client
+
+**Time Beetle**: "You can also create a client that connects to the MCP server and interacts with it. Here's an example of how to create a simple MCP client using the `@modelcontextprotocol/sdk` library:
+
+```typescript
+import { Client } from "@modelcontextprotocol/sdk/client/index.js";
+import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";
+
+const transport = new StdioClientTransport({
+  command: "node",
+  args: ["server.js"]
+});
+
+const client = new Client(
+  {
+    name: "example-client",
+    version: "1.0.0"
+  }
+);
+
+await client.connect(transport);
+```
+
+In the preceding code, we:
+
+- Created a transport layer using the `StdioClientTransport` class, which allows the client to communicate with the MCP server through standard input and output.
+- Initiated a new `Client` instance with a name and version.
+- Connected the client to the transport layer using the `connect` method.
+
+**You**: "Let me guess, there's a lot more we can do with the client?"
+
+**Time Beetle**: Yes, let's talk about that next. The MCP client can be used to interact with the server, call tools, and access resources. 
+
+### Listing and calling tools and resources
+
+**Time Beetle**: "There's generally two scenarios you want to cover, listing tools and resources, and calling them. Here's how to do that:
+
+```typescript
+// List prompts
+const prompts = await client.listPrompts();
+
+// List resources
+const resources = await client.listResources();
+
+// list tools
+const tools = await client.listTools();
+```
+
+**You**: This is great, then I know what capabilities the server has. But how do I call them?
+
+**Time Beetle**: "Right, let me dive into a specific example, tools first. So you start by asking for what tools you have, then you can store that response and then call the tool you want. Here's an example:
+
+So when you list tools, you get a response on this format:
+
+```json
+  
+  {
+    "name": "<tool name>",
+    "description": "<description>".
+    "inputSchema": {
+       "type":"object",
+       "properties":{
+          "a":{
+            "type":"number"
+          },
+          "b":{
+            "type":"number"
+          }},
+          "required":["a","b"],
+          "additionalProperties":false,
+          "$schema":"http://json-schema.org/draft-07/schema#"
+      }
+  }
+```
+
+which means, if you have the tools `add` and `subtract`, your response looks like so:
```

Comment:
The code block below probably needs to be moved up so it's right below this. Otherwise it's confusing since "looks like so:" is followed by another narrative rather than the code.